src/api/admin/mermaid.py

```diff
@@ -549,28 +555,34 @@ def fk_link(self, obj):
     def region_list(self, obj):
         return ",".join([r.name for r in obj.regions.all()])
 
+    def life_history_list(self, obj):
+        return ",".join([lh.name for lh in obj.life_histories.all()])
+
     def get_readonly_fields(self, request, obj=None):
         if obj:  # editing an existing object
-            return ("child_life_histories", "child_regions")
+            return ("child_regions", "child_life_histories")
         return ()
 
     def child_life_histories(self, obj):
         proportions = {}
-        # get all non-null life histories of descendants
-        life_histories = [ba.life_history for ba in obj.descendants if ba.life_history]
-        total = float(len(life_histories))
-        for lh in life_histories:
-            if lh.name not in proportions:
-                proportions[lh.name] = 0
-            proportions[lh.name] += 1 / total
+        counts = {}

```

Comment:
Could use `counts = defaultdict(int)` and then get rid `if lh.name not in counts:`

Reply:
refactored